packages/vite/src/node/server/middlewares/base.ts

```diff
@@ -29,18 +30,18 @@ export function baseMiddleware({
     if (path === '/' || path === '/index.html') {
       // redirect root visit to based url with search and hash
       res.writeHead(302, {
-        Location: devBase + (parsed.search || '') + (parsed.hash || '')
+        Location: config.base + (parsed.search || '') + (parsed.hash || '')
       })
       res.end()
       return
     } else if (req.headers.accept?.includes('text/html')) {
       // non-based page visit
-      const redirectPath = devBase + url.slice(1)
+      const redirectPath = joinUrlSegments(config.base, url.slice(1))
```

Comment:
We no longer need the `.slice(1)` here, no? (There are two other examples like this)

Reply:
good call. I've updated it throughout